cluster/setup.go

```diff
@@ -0,0 +1,191 @@
+// This file is part of MinIO Kubernetes Cloud
+// Copyright (c) 2019 MinIO, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package cluster
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/golang-migrate/migrate/v4"
+	// the postgres driver for go-migrate
+	_ "github.com/golang-migrate/migrate/v4/database/postgres"
+	// the file driver for go-migrate
+	_ "github.com/golang-migrate/migrate/v4/source/file"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/api/extensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	m3SystemNamespace = "m3"
+)
+
+// Setups m3 on the kubernetes deployment that we are installed to
+func SetupM3() {
+	fmt.Println("Setting up m3 namespace")
+	SetupM3Namespace()
+	fmt.Println("Setting up postgres")
+	SetupPostgres()
+	fmt.Println("Running Migrations")
+	RunMigrations()
+}
+
+// Setups a postgres used by the provisioning service
+func SetupM3Namespace() {
+	config := getConfig()
+	// creates the clientset
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	namespace := v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "m3",
+		},
+	}
+
+	_, err = clientset.CoreV1().Namespaces().Create(&namespace)
+	if err != nil {
+		fmt.Println(err)
+	}
+}
+
+// Setups a postgres used by the provisioning service
+func SetupPostgres() {
+
+	config := getConfig()
+	// creates the clientset
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	pgSvc := v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "postgres",
+			Labels: map[string]string{
+				"name": "postgres",
+			},
+		},
+		Spec: v1.ServiceSpec{
+			Ports: []v1.ServicePort{
+				{
+					Name: "http",
+					Port: 5432,
+				},
+			},
+			Selector: map[string]string{
+				"app": "postgres",
+			},
+		},
+	}
+
+	res, err := clientset.CoreV1().Services(m3SystemNamespace).Create(&pgSvc)
+	if err != nil {
+		panic(err.Error())
+	}
+	fmt.Println("done setting up postgres servcice ")
+	fmt.Println(res.String())
+
+	configMap := v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "postgres-env",
+			Labels: map[string]string{
+				"app": "postgres",
+			},
+		},
+		Data: map[string]string{
```

Comment:
I think for postgres credentials using a kubernetes secret instead of kubernetes configMap is better but its good for now, we can change that later.

Reply:
Yes, we have to move to secrets eventually, we can do that on the next PR